### PR TITLE
[SPARK-29711][SQL] Support dynamic adjust sql class log level

### PR DESCRIPTION
**Pull request by deshanxiao**
### What changes were proposed in this pull request?

Support dynamic change the log level in sql

### Why are the changes needed?
It is very convenient for us to debug some problem about sql. We will not restart the thriftserver to see some debug logs. And this change is very small.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
UT

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -380,6 +380,15 @@ class SparkContext(config: SparkConf) extends Logging {
     Utils.setLogLevel(org.apache.log4j.Level.toLevel(upperCased))
   }
 
+  def setLogLevel(logLevel: String, className: String): Unit = {
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: I am not sure about this one. `setLogLevel` is basically just for quick debugging purpose in a shell in general. If we want some more complicated log settings, it should go for log4j.

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala [hunk 2]
```diff
@@ -372,6 +374,16 @@ private[hive] class SparkExecuteStatementOperation(
       }
     }
   }
+
+  private def prepareLogLevel(): Unit = {
+    val debugClasses = sqlContext.conf.debugClasses()
```
Comment (HyukjinKwon) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala]: this configuration seems only affecting thriftserver. it should have an effect in regular Spark session as well. But more importantly, why don't you just restart or start another one for debugging purpose? I don't think this configuration is worth much.

Comment (deshanxiao) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala]: We just want to change the log level of thriftserver when our application has already deployed. If it doesn't support dynamic log level, we should change the log4j file and restart the thrift server every time. It is not convenient.

File: core/src/main/scala/org/apache/spark/SparkContext.scala (hunk 1 above)
Comment (deshanxiao) [reply core/src/main/scala/org/apache/spark/SparkContext.scala]: Yes, I just added the method to support thriftserver dynamic log level. It also looks great to support shell debugging.

**Pull request closed without merging**